internal/ingress/annotations/authreq/main_test.go

```diff
@@ -243,6 +243,66 @@ func TestCacheDurationAnnotations(t *testing.T) {
 	}
 }
 
+func TestKeepaliveAnnotations(t *testing.T) {
+	ing := buildIngress()
+
+	data := map[string]string{}
+	ing.SetAnnotations(data)
+
+	tests := []struct {
+		title                string
+		url                  string
+		keepaliveConnections string
+		keepaliveRequests    string
+		keepaliveTimeout     string
+		expectedConnections  int
+		expectedRequests     int
+		expectedTimeout      int
+	}{
+		{"all set", "http://goog.url", "5", "500", "50", 5, 500, 50},
+		{"no annotation", "http://goog.url", "", "", "", defaultKeepaliveConnections, defaultKeepaliveRequests, defaultKeepaliveTimeout},
+		{"default for connections", "http://goog.url", "x", "500", "50", defaultKeepaliveConnections, 500, 50},
+		{"default for requests", "http://goog.url", "5", "x", "50", 5, defaultKeepaliveRequests, 50},
+		{"default for invalid timeout", "http://goog.url", "5", "500", "x", 5, 500, defaultKeepaliveTimeout},
+		{"variable in host", "http://$host:5000/a/b", "5", "", "", 0, defaultKeepaliveRequests, defaultKeepaliveTimeout},
+		{"variable in path", "http://goog.url:5000/$path", "5", "", "", 5, defaultKeepaliveRequests, defaultKeepaliveTimeout},
```

Comment:
shoud we add case for the values less than 0?

Reply:
I haven't seen any tests for checking values less than 0 or handling negative values in other int annotations. Eg. `auth-tls-verify-depth`, `cors-max-age`, `global-rate-limit`, `proxy-connect-timeout`, `proxy-buffers-number`, `proxy-ssl-verify-depth`, `limit-rate-after`, ...
Should I add tests for values less than 0?

Reply:
@leki75 I think we should add. Some scenarios have not been covered yet, and I think some of them need to be added.

Reply:
Done